### PR TITLE
28 Changelog

### PR DESCRIPTION
**Pull request by ruchim**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,52 @@
...
+task writer {
+  Array[String] a = ["foo", "bar"]
+  # used to output: "foo\nbar"
+  # now outputs: "foo\nbar\n"
+  File out = write_lines(a)
+  ...
+}
+```
+
+* A workflow utilizing the WorkflowFailureMode Workflow Option `ContinueWhilePossible` will now successfully reach a terminal state once all runnable jobs have completed.
```
Comment (mcovarr) [inline CHANGELOG.md]: If the `write_*` stuff gets four octothorpes I would think this would too

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,52 @@
...
+
+#### WDL write_* functions add a final newline
+
+The following WDL functions now add a newline after the final line of output (the previous behavior of not adding this
+newline was inadvertent):
+- write_lines
+- write_map
+- write_object
+- write_objects
+- write_tsv
```
Comment (cjllanwarne) [inline CHANGELOG.md]: These could be backticked

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,52 @@
...
+
+For example:
+
+```
+task writer {
+  Array[String] a = ["foo", "bar"]
+  # used to output: "foo\nbar"
+  # now outputs: "foo\nbar\n"
+  File out = write_lines(a)
```
Comment (cjllanwarne) [inline CHANGELOG.md]: I'd prefer it if this was more realistic, something like:
```
task writer {
  Array[String] a = ["foo", "bar"]
  command {
    # used to output: "foo\nbar"
    # now outputs: "foo\nbar\n"
    cat write_lines(a)
  }
}
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,52 @@
...
+
+### WDL Functions
+
+#### Floor, ceil and round:
+
+Enables the `floor`, `ceil` and `round` functions in WDL to convert floating point numbers to integers.
+
+For example we can now use the size of an input file to influence the amount of memory the task is given. In the example below a 500MB input file will result in a request for a VM with 2GB of memory:
+
+```task foo {
```
Comment (cjllanwarne) [inline CHANGELOG.md]: Need newline between backticks and task. Right now this line is unrendered

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,52 @@
...
+  # used to output: "foo\nbar"
+  # now outputs: "foo\nbar\n"
+  File out = write_lines(a)
+  ...
+}
+```
+
+* A workflow utilizing the WorkflowFailureMode Workflow Option `ContinueWhilePossible` will now successfully reach a terminal state once all runnable jobs have completed.
+* If `FailOnStderr` is set to false, Cromwell no longer checks for the existence of a stderr file for that task. 
+* General improvements to error reporting for workflow failures.
```
Comment (mcovarr) [inline CHANGELOG.md]: not sure why this sentence is here

File: CHANGELOG.md [hunk 6]
```diff
@@ -23,7 +69,10 @@ See the [README](https://github.com/broadinstitute/cromwell#get-apiworkflowsvers
 * The workflow submission parameters `wdlSource` and `wdlDependencies` have been deprecated in favor of `workflowSource` and
 `workflowDependencies` respectively.  The older names are still supported in Cromwell 28 with deprecation warnings but will
 be removed in a future version of Cromwell.
+
+### Labels
 * A new `/labels` endpoint has been added to update labels for an existing workflow. See the [README](README.md#patch-apiworkflowsversionidlabels) for more information.
+* Label formatting requirements have been updated, please check the [README](README.md#label-format) for more detailed documentation.
```
Comment (cjllanwarne) [inline CHANGELOG.md]: Is this a breaking change?

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,6 +2,52 @@
...
+}
+```
+
+* A workflow utilizing the WorkflowFailureMode Workflow Option `ContinueWhilePossible` will now successfully reach a terminal state once all runnable jobs have completed.
+* If `FailOnStderr` is set to false, Cromwell no longer checks for the existence of a stderr file for that task. 
+* General improvements to error reporting for workflow failures.
+
+### WDL Functions
+
+#### Floor, ceil and round:
```
Comment (mcovarr) [inline CHANGELOG.md]: `floor` should probably not be capitalized

File: CHANGELOG.md [hunk 8]
```diff
@@ -55,27 +97,13 @@ filesystems {
...
-- write_lines
-- write_map
-- write_object
-- write_objects
-- write_tsv
+### Metadata Write Batching
 
-For example:
+Metadata write batching works the same as in previous versions of Cromwell, but the default batch size has been changed from 
+a default of 1 to 200.  It's possible that 200 is too high in some environments, but more likely to be an appropriate value
```
Comment (mcovarr) [inline CHANGELOG.md]: "a default of" => ""

(the sentence already says "default" earlier)

File: CHANGELOG.md (hunk 5 above)
Comment (ruchim) [inline CHANGELOG.md]: I wanted to salute your new error reporting for invalid host names

File: CHANGELOG.md (hunk 7 above)
Comment (cjllanwarne) [inline CHANGELOG.md]: 😨 but it starts a sentence!

File: CHANGELOG.md (hunk 5 above)
Comment (mcovarr) [inline CHANGELOG.md]: Oh, then it may need octothorpes and probably more detail too.  🙂

File: CHANGELOG.md (hunk 6 above)
Comment (ruchim) [inline CHANGELOG.md]: The rules for label values now have more room for variation (i.e, they can be empty strings or start with a number). Whatever custom labels someone was submitting previously should still be valid.

File: CHANGELOG.md (hunk 7 above)
Comment (mcovarr) [inline CHANGELOG.md]: backticks then?

File: CHANGELOG.md [hunk 9]
```diff
@@ -55,27 +101,12 @@ filesystems {
...
-- write_lines
-- write_map
-- write_object
-- write_objects
-- write_tsv
+### Metadata Write Batching
 
-For example:
+Metadata write batching works the same as in previous versions of Cromwell, but the default batch size has been changed from 1 to 200.  It's possible that 200 is too high in some environments, but more likely to be an appropriate value
+than the previous default.
```
Comment (mcovarr) [inline CHANGELOG.md]: but more => but is more

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**